### PR TITLE
Remove deprecated petsc things

### PR DESCRIPTION
**Pull request by drwells**
These were deprecated in #4341 - it's been four years, its time to say goodbye.

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

File: tests/petsc/64.cc [hunk 1]
```diff
@@ -67,8 +70,18 @@ main(int argc, char **argv)
         const unsigned int n_mpi_processes = static_cast<unsigned int>(n_jobs);
         Assert(n_dofs % n_mpi_processes == 0, ExcInternalError());
         const unsigned int n_local_dofs = n_dofs / n_mpi_processes;
```
Comment (marcfehling) [inline tests/petsc/64.cc:72]: You don't use `n_local_dofs` anywhere in this test. Is that intentional?

**Review by marcfehling — COMMENTED** [review]
The test `lac/linear_operator_09` fails after this change. It still uses the old constructor:
https://github.com/dealii/dealii/blob/c7680c2d49b55ba7fde98a0fb7ca90e2c53bcab6/tests/lac/linear_operator_09.cc#L48-L49

File: tests/petsc/64.cc (hunk 1 above)
Comment (drwells) [reply tests/petsc/64.cc:72]: No - that should be removed.

**Comment by drwells** [comment]
Thanks for the heads up - I'll put up a followup PR to fix the other tests later.

**Pull request merged**